js/adapt-vimeo.js

```diff
@@ -1,146 +1,147 @@
-
-      if (transcriptConfig._setCompletionOnView) {
-        this.listenToOnce(this.transcriptView, 'transcript:open', this.setCompletionStatus);
-      }
-    },
-
-    setupEventListeners: function() {
-      this.listenTo(this.vimeoView, {
-        play: this.handleMediaEvent,
-        pause: this.handleMediaEvent,
-        ended: this.handleMediaEvent
-      });
-
-      var completionEvent = COMPLETION(this.model.get('_setCompletionOn')) || COMPLETION.PLAY;
-
-      if (completionEvent === COMPLETION.INVIEW) {
-        return this.setupInviewCompletion('.component__widget', this.setCompletionStatus);
-      }
-
-      this.listenToOnce(this.vimeoView, completionEvent.asLowerCase, this.setCompletionStatus);
-    },
-
-    /**
-     * Add a sub view to this component's widget element
-     * @param {Backbone.View} constructor
-     * @param {object} model - the data to pass as the view's model
-     * @returns {Backbone.View}
-     */
-    addSubview: function(constructor, model) {
-      var view = new constructor({ model: model });
-      this.$widget.append(view.$el);
-      return view;
-    },
-
-    onSkipToTranscript: function() {
-      Adapt.a11y.focusFirst(this.$('.vimeo__transcript-btn'));
-    },
-
-    /**
-     * Filter rapidly firing events so that only meaningful 'media' events are triggered
-     * @param {object} event
-     */
-    handleMediaEvent: function(event) {
-      var eventType = event.type;
-      var rapidFireTimeoutWindow = 500;
-
-      var currentTime = Date.now();
-      var lastEventTime = this.lastEventTime || 0;
-      this.lastEventTime = currentTime;
-
-      clearTimeout(this.waitingTriggerTimeout);
-
-      var isAlwaysTriggered = (eventType === 'ended');
-      if (isAlwaysTriggered) {
-        return this.triggerGlobalEvent(eventType);
-      }
-
-      var lastEventDuration = (currentTime - lastEventTime);
-      var isRapidlyFiring = (lastEventDuration <= rapidFireTimeoutWindow);
-      if (isRapidlyFiring) return;
-
-      // Wait for rapidFireTimeoutWindow until triggering
-      // Allows trigger to be dropped if another incoming event arrives
-      var trigger = this.triggerGlobalEvent.bind(this, eventType);
-      this.waitingTriggerTimeout = setTimeout(trigger, rapidFireTimeoutWindow);
-    },
-
-    /**
-     * Trigger a global 'media' event
-     * @param {string} eventType
-     */
-    triggerGlobalEvent: function(eventType) {
-      Adapt.trigger('media', {
-        isVideo: true,
-        type: eventType,
-        src: this.model.get('_media')._source,
-        platform: 'Vimeo'
-      });
-    },
-
-    /**
-     * Remove this view and sub views. Destroys the vimeo player instance
-     */
-    remove: function() {
-      this.vimeoView.remove();
-      this.transcriptView && this.transcriptView.remove();
-      ComponentView.prototype.remove.call(this);
+    };
+  }
+
+  preRender() {
+    this.COMPLETION = ENUM([
+      'INVIEW',
+      'PLAY',
+      'ENDED'
+    ], value => {
```

Comment:
These are completion triggers/events, rather than states as per the enum in core.

Suggest the name is changed to reflect this e.g. `COMPLETION` or `COMPLETION_EVENT`.